src/plugins/condition/criterion/TelemetryCriterion.js

```diff
@@ -44,33 +44,36 @@ export default class TelemetryCriterion extends EventEmitter {
         this.operation = telemetryDomainObjectDefinition.operation;
         this.input = telemetryDomainObjectDefinition.input;
         this.metadata = telemetryDomainObjectDefinition.metadata;
-        this.subscription = null;
-        this.telemetryObjectIdAsString = null;
-        this.objectAPI.get(this.objectAPI.makeKeyString(this.telemetry)).then((obj) => this.initialize(obj));
+        this.telemetryObjectIdAsString = this.objectAPI.makeKeyString(this.telemetry);
+        this.on(`subscription:${this.telemetryObjectIdAsString}`, this.handleSubscription);
+        this.objectAPI.get(this.telemetryObjectIdAsString).then((obj) => this.initialize(obj));
```

Comment:
This change to use `this.telemetryObjectIdAsString` instead of `this.telemetry` will fail hard when you add a criterion as the telemetry is not yet set.
Either revert this change or ensure that a criterion class isn't created unless the telemetry has been chosen for it.

Reply:
Makes sense. (working backwards in replying to comments...) This was done because this.telemetryObject was going to be removed as not necessary for providing telemetry. If the telemetry object is still needed I can revert this.

Reply:
Reverting to have telemetry object and setting in initialize function.

Reply:
Reverted to use telemetryObject and initialize from that